routers/api/v1/repo/tag.go

```diff
@@ -33,7 +33,7 @@ func ListTags(ctx *context.APIContext) {
 	// responses:
 	//   "200":
 	//     "$ref": "#/responses/TagList"
-	tags, err := ctx.Repo.Repository.GetTags()
+	tags, err := ctx.Repo.GitRepo.GetTagInfos()
```

Comment:
So each `tags[]` element will contain a reference to the same `Repository` object, and it should be the same as `ctx`'s, so hopefully the `ctx.Repository.Close()` at the end/defer of the ctx should be enough to ensure that all files are closed. 🤔

Reply:
Yes. Any packfiles are kept open in the gogitStorage within the repo. Nowhere else as far as I can see.

Reply:
Why this should be changed here?

Reply:
Because otherwise it's impossible to cleanly close the gogit.Repository.